fix: Treat Document Links entries as all non-std fields

marination opened the pull request:
## Issue
- Add a document link via Customize Form with a non standard fieldname. [Here I added the following to **Employee**]
  <img width="900" alt="Screenshot 2023-11-24 at 5 13 28 PM" src="https://github.com/frappe/frappe/assets/25857446/0cab1ce1-c8ed-4a04-a220-36e9c531de35">
- The doctype has a custom dashboard override via the hooks
-  The dashboard uses the wrong fieldname to get the custom link data
   <img width="900" alt="Screenshot 2023-11-24 at 5 16 01 PM" src="https://github.com/frappe/frappe/assets/25857446/d7160d05-1a3a-4174-a928-80f1bd790b21">

## Why it breaks
- The doctype links are fetched and then the hooks overrides are added
- `meta.py` tries to "guess" the standard fieldname from the document links. In this case it assumes **party** is the standard fieldname (because it is the first entry). Thus **it is never added to non standard fieldnames**
- Then, the custom dashboard, in this case, via hooks overrides the standard fieldname to `employee`
- Since **Bank Transaction** has no non standard fieldname, standard fieldname  `employee` is used against it > hence the error

## What fixes it:
- Since the field name is specified/mandatory in the Document Links table > use it as a non standard field because it is hard to determine which field is standard. And we honestly don't need the guess work afaik
   <img width="800" alt="Screenshot 2023-11-24 at 5 22 28 PM" src="https://github.com/frappe/frappe/assets/25857446/2ccee20b-9a8e-4b6c-afd8-4be67cc6c79e">

marination committed:
fix: Treat Document Links entries as all non-std fields

- Since the field name is specified use it as a non standard field because it is hard to determine which field is standard. It is often wrong to assume the first entry has the statndard field

marination committed:
test: Custom doctype links having non std fieldname with dashboard override

marination committed:
style: Linter

akhilnarang commented:
[This](https://github.com/frappe/frappe/blob/58b75d7c35794edb8a4ec8df06adcbe9f79d74c4/frappe/custom/doctype/customize_form/test_customize_form.py#L239) test is failing now ([here](https://github.com/frappe/frappe/blob/58b75d7c35794edb8a4ec8df06adcbe9f79d74c4/frappe/custom/doctype/customize_form/test_customize_form.py#L256))

marination commented:
@akhilnarang I couldn't replicate it locally, does the test seem related to the fix ?

akhilnarang commented:
> @akhilnarang I couldn't replicate it locally, does the test seem related to the fix ?

Hmm, seems to fail anytime its run twice on the same DB (even on develop branch)
So probably happening here because the new test is also similar.

akhilnarang committed:
fix(test_customize_form): drop checking of `flags.update_db`

The value changed is being checked directly
Checking the flag breaks if this or a similar test is run multiple times on the same database

Also extract 255 into a separate variable

Signed-off-by: Akhil Narang <me@akhilnarang.dev>
